apps/newsletters-ui/src/app/components/ArrayInput.tsx

```diff
@@ -32,36 +42,63 @@ export const ArrayInput = ({
 	};
 
 	return (
-		<fieldset>
-			<legend>
-				{label} ({data.length} items)
-			</legend>
-			{validationWarning && <b>!{validationWarning}</b>}
+		<Box padding={2} border={1} borderColor={''} borderRadius={1}>
+			<Badge badgeContent={data.length} color="primary">
+				<Typography sx={{ fontWeight: 700, fontSize: 12 }}>{label}</Typography>
+			</Badge>
+
+			<Grid container alignItems={'center'} rowSpacing={1} columnSpacing={2}>
+				{data.length === 0 && (
+					<Grid item xs={12}>
+						<Alert severity="info">No Items</Alert>
+					</Grid>
+				)}
+
+				{data.map((item, index) => (
+					<Fragment key={index}>
+						<Grid item xs={9}>
+							<TextField
+								variant="standard"
+								fullWidth
+								value={item}
+								onChange={(event) => handleInput(event, index)}
+							/>
+						</Grid>
+						<Grid item xs={3}>
+							<Button
+								size="small"
+								color="error"
+								variant="outlined"
+								title={`delete entry "${item}"`}
+								onClick={() => {
+									handleDelete(index);
+								}}
+							>
+								x
+							</Button>
+						</Grid>
+					</Fragment>
```

Comment:
React added <></> as a more modern update to fragments. Just another way to skin the same rabbit.